src/mlpack/core/util/param_checks_impl.hpp

```diff
@@ -306,6 +306,29 @@ inline void ReportIgnoredParam(util::Params& params,
   }
 }
 
+// Check if the given input data points aren't empty.
+inline void RequireNonEmptyInputValue(
+    util::Params& params,
+    const std::string& paramName,
+    const bool fatal,
+    const std::string& errorMessage)
+{
+  if (BINDING_IGNORE_CHECK(paramName))
+    return;
+
+  if (params.Has(paramName))
+  {
+    arma::mat trainingData = std::move(params.Get<arma::mat>(paramName));
```

Comment:
This may not always be an `arma::mat` parameter.  It could also be `arma::Mat<size_t>`, `arma::Col<size_t>`, `arma::colvec`, etc.  You might want to take the type of the parameter as a template argument.

Reply:
I used it as a template parameter instead of ``arma::mat``.